File: solutions/014.longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let strsLength = strs.length
    if(strsLength === 0)
        return ''
    for(let i = 0;;i++){
        for(let j =0;j<strsLength;j++){
            let item =''
            if(!strs[j][i]){
                return strs[j]
            }else{
                if(strs[j][i]!==strs[0][i]){
                    return strs[0].slice(0,i)
                  
                }
            }
        }
       
      
        
    }
    
};

File: solutions/014.longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int length = strs.length;
        int end = -1;
        Boolean hasBreaked = false;
        if(length==0){
            return "";
        }else if(length == 1){
            return strs[0];
        }else{
            for(int i =0;i<strs[0].length();i++){
          
                
                for(int j = 1;j<length;j++){
                   try{
                    if(strs[j].charAt(i)!=strs[0].charAt(i)){
                        hasBreaked = true;
                        break;
                    }
                }catch(Exception e){
                    hasBreaked = true;
                    break;
                } 
                }
                
                if(hasBreaked){
                    break;
                }else{
                    end = i+1;
                }
            }
        }
        if(end == -1){
            return "";
        }else if(end >= strs[0].length()){
            return strs[0];
        }
        return strs[0].substring(0,end);
        
        
        
    }
}